website/ref/language.md

```diff
@@ -102,27 +104,79 @@ metacharacters; use quoted strings instead. For instance, to echo a star, write
 in line continuations; their use elsewhere is reserved will cause a syntax
 error.
 
-## Notes
+### Notes
 
 The three syntaxes above all evaluate to strings, and they are interchangeable.
 For instance, `xyz`, `'xyz'` and `"xyz"` are different syntaxes for the same
-string, and they are always equivalent.
+string, and they are always equivalent with the exception of
+**escape sequences** as documented above.
+
+## Number
+
+Elvish has a numeric data type, `float64`, created using this syntax:
+`(float64 $number)`. For convenience, all Elvish functions that
+expect a numeric argument will accept a string that can be parsed as a
+number. They will automatically convert the string to a `float64`. Which
+means explicit `float64` constructors are almost never needed. Not even
+for special values like infinity (`Inf`) or not-a-number (`NaN`). Note
+that the letter case of these magic values doesn't matter so `inf` and
+`Inf` are equivalent.
+
+The `$number` argument can be a string in the following formats:
+
+* decimal notation (e.g., `10`),
+* hexadecimal notation (e.g., `0xA`),
+* octal notation (e.g., `0o12`),
+* binary notation (e.g., `0b1010`),
+* floating point notation (e.g., `10.0`), and
+* scientific notation (e.g., `1.0e1`).
+
+Note that the examples in the above list all express the same value.
+
+To convert a `float64` variable to a string do `(to-string $number)`
+which is equivalent to `(printf '%.16f' $number)`.  Alternatively,
+explicitly format it yourself with the desired representation; e.g.,
+`(printf '%.3f' $number)`.
 
-Elvish does have a dedicated `float64` number type, but it does not have a
-literal syntax; instead it needs to be constructed using a function call, as in
-`put (float64 42)`; the argument `42` is a string. For convenience, all Elvish
-functions that expect number arguments also accept strings that can be parsed as
-a number, so explicit number constructors are not needed often.
+## Exception
 
+Elvish has an exception data type, but it does not have a literal
+syntax for that type (despite the literal representation shown
+below by `put`). Instead it needs to be constructed using the
```

Comment:
> Instead it needs to be constructed using the fail command: ...

This is technically correct. However, since this is the *language* doc, the text should focus on the important language construct related to exceptions. In this case, the important language construct is [exception capture](https://elv.sh/ref/language.html#exception-capture); `fail` is *not* a fundamental part of the language, it just happens to be a builtin that always throws an exception.

Also, there is already some doc on exceptions in [another section](https://elv.sh/ref/language.html#exception-and-flow-commands); better not to repeat things.

I suggest moving the exception section to a different PR; the documentation improvements for the number type is much more solid and can be merged soon.

Reply:
Okay, but I think it is important for all data types, including exceptions, to be documented in this major section. What I'll do is turn this minor section into a stub that refers the reader to https://elv.sh/ref/language.html#exception-and-flow-commands for a broader discussion of how exceptions work.